scylla/src/transport/topology.rs

```diff
@@ -905,11 +922,15 @@ async fn create_peer_from_row(
     }))
 }
 
-fn query_filter_keyspace_name<'a>(
+fn query_filter_keyspace_name<'a, R>(
     conn: &Arc<Connection>,
     query_str: &'a str,
     keyspaces_to_fetch: &'a [String],
-) -> impl Stream<Item = Result<Row, QueryError>> + 'a {
+    convert_typecheck_error: impl FnOnce(TypeCheckError) -> MetadataError + 'a,
+) -> impl Stream<Item = Result<R, QueryError>> + 'a
+where
+    R: for<'r> DeserializeRow<'r, 'r> + 'static,
+{
```

Comment:
`FnOnce` unfortunately cannot be used with `&dyn` - calling `FnOnce` need ownership of it. I'll investigate using `&dyn Fn` here.

Reply:
Actually, `impl FnOnce` does not hurt us, because `R` makes this function monomorphised anyway. I made an optimisation, though: I degenerised a large chunk of this function, so that the monomorphisation penalty is smaller.